app/views/posts/index.html.erb

```diff
@@ -1 +1,49 @@
-<h1>show a list of posts</h1>
\ No newline at end of file
+<div class = 'users-tab'>
+  <h1>All Posts</h1>
+  <div>
+    <div>
+    <div class = 'img-container'>
+          <div class="inner-border">
+            <img src="<%= @posts.first.author.photo %>" alt="" width="100" height="100">
+          </div>
+        </div>
+      
+      <%= link_to @posts.first.author.name, user_path(@posts.first.author) %>
+    </div>
+    <div class='posts-list'>
+    <% @posts.each do |post| %>
+      <div class="posts-index">
+        <div class="posts-title">
+          <h3>
+            <%# author id: <%= post.author_id %>
+            <%= link_to post.title, user_post_path(post.author_id, post.id) %> 
+          </h3>
+        </div>
+        <div class="posts-comment">
+          <h4>
+            <% if post.comments.empty? %> 
+              No comments yet
+            <% else %>
+              <ul>
+                <% post.comments.each do |comment| %>
+              </ul>
+              <li>
+                <%= comment.author.name %>:
+                <%= comment.text %>
+              </li>
+                <% end %>
+            <% end %>

```

Comment:
- [ ] According to the requirements, in this part of the code you should only get the five most recent comments on that post.

Reply:
thank you for your review @mikethreels, but I think that according to the requirements the index must to show all the posts, and the recent post must be used when you enter the profile of the user. And they are 3 not 5 recent post. You can see that in the previous requirements of the project, [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md)